from PyQt5.uic import loadUi

from PyQt5.QtWidgets import QWidget

from views.misc.messagebox import MessageBox


class ResumeView(QWidget):
    def __init__(self, resumeController, parent):
        super(ResumeView, self).__init__(parent)
        self.resumeController = resumeController
        loadUi('./views/ui/widgets/resume.ui', self)

        # Connect thing
        self.comboBoxRouter.addItem("Huawei")
        self.comboBoxRouter.currentIndexChanged.connect(self.getModels)
        # Set verticalLayout. It's resize the widget
        self.setLayout(self.verticalLayout)

    def setInformationProject(self, data):
        self.labelNameFile.setText(data[0])
        self.labelIPFile.setText(data[1])
        self.labelMACFile.setText(data[2])
        self.labelOUIFile.setText(data[3])
        self.labelCompanyFile.setText(data[4])

    def setModels(self, models):
        if models:
            for model in models:
                self.comboBoxModel.addItem(str(model))
        else:
            MessageBox.critical("Error", "We can not found models for this router. Try to choose other router.")

    def getModels(self):
        self.comboBoxModel.clear()
        self.resumeController.getModelsForRouter(self.getRouterActive())

    def getRouterActive(self):
        router = self.comboBoxRouter.currentText()
        if "None" in router:
            return None
        else:
            return router

    def getModelActive(self):
        return self.comboBoxModel.currentText()
